jetbrains-ultimate/src-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftUtils.kt

```diff
@@ -0,0 +1,17 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.redshift
+
+import com.intellij.openapi.project.Project
+import software.amazon.awssdk.services.redshift.model.Cluster
+import software.aws.toolkits.core.region.AwsRegion
+import software.aws.toolkits.core.utils.tryOrNull
+import software.aws.toolkits.jetbrains.core.AwsResourceCache
+import software.aws.toolkits.jetbrains.services.sts.StsResources
+
+fun Project.clusterArn(cluster: Cluster, region: AwsRegion): String {
```

Comment:
We should probably make a generic 'arn' util, I could see something like:

```kt
Project.arn(
    serviceName: String, 
    resourceType: String, 
    resourceIdentifier: String, 
    connection: ConnectionSettings? = null
)
```

Could be something we do in the future tho - could take a leaf out of the CDK book : https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/core/lib/arn.ts

Reply:
Yep I mostly included this to get comments, we have to do the same thing for S3 (and probably others) so a util in master would make sense